warehouse/query-core/src/test/java/datawave/query/language/parser/jexl/JexlNodeSetTest.java

```diff
@@ -0,0 +1,266 @@
+package datawave.query.language.parser.jexl;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import datawave.query.jexl.JexlASTHelper;
+import datawave.query.jexl.JexlNodeFactory;
+import org.apache.commons.jexl2.parser.JexlNode;
+import org.junit.Test;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class JexlNodeSetTest {
+    
+    @Test
+    public void testTheReasonForThisClass() {
+        JexlNode node1 = JexlNodeFactory.buildEQNode("FOO", "bar");
+        JexlNode node2 = JexlNodeFactory.buildEQNode("FOO", "bar");
+        
+        Set<JexlNode> nodeSet = Sets.newHashSet(node1, node2);
+        assertEquals(2, nodeSet.size());
+        assertTrue(nodeSet.contains(node1));
```

Comment:
Again, not to harp on it, but we are getting this behavior b/c of the default implementation of `equals()` and `hashCode()`. This test would fail if we overrode those to use your `nodeToKey` function.